python_modules/dagster/dagster/_core/execution/build_resources.py

```diff
@@ -124,6 +130,10 @@ def wrap_resources_for_execution(
     for resource_key, resource in resources.items():
         if isinstance(resource, ResourceDefinition):
             resource_defs[resource_key] = resource
+        elif is_structured_resource_cls(resource):
+            resource_defs[resource_key] = UnconfiguredStructuredResource(resource)
+        elif is_structured_io_manager_cls(resource):
+            resource_defs[resource_key] = UnconfiguredStructuredIOManager(resource)
```

Comment:
can we move this to a structured resource-only codepath?

Reply:
Could you elaborate on what you mean? The `is_structured_X_cls` functions do restrict these statements to structured resources only. Did you mean Definitions-only codepath?